xline/src/server/lease_server.rs

```diff
@@ -324,7 +328,11 @@ where
                 };
                 return Ok(tonic::Response::new(res));
             }
-            let leader_id = self.client.get_leader_id_from_curp().await;
+            let leader_id = self
+                .client
+                .get_leader_id_from_curp()
+                .await
+                .map_err(|e| tonic::Status::internal(format!("Get leader id error: {e}")))?;
```

Comment:
`propose_err_to_status` convert `CommandProposeError` to `tonic::Status`, not `ProposeError`

Reply:
So why don't we convert the `ProposeError` to the `CommandProposeError` and then use `propose_err_to_status`?  IMO, our current error-handling logic is somewhat complex. It's better to unify it using a single method for error transformation, as it will be less prone to oversight during the subsequent refactoring process.